packages/ui/components/ui/tx/view/viewbox.tsx

```diff
@@ -7,22 +7,25 @@ import { IncognitoIcon } from '../../icons/incognito';
 export interface ViewBoxProps {
   label: string;
   visibleContent?: React.ReactElement;
+  isOpaque?: boolean;
 }
 
 const Label = ({ label }: { label: string }) => <span className='text-lg'>{label}</span>;
 
-export const ViewBox = ({ label, visibleContent }: ViewBoxProps) => {
+export const ViewBox = ({ label, visibleContent, isOpaque }: ViewBoxProps) => {
+  // if isOpaque is undefined, set it to !visibleContent
```

Comment:
Hm, couldn't this conflate private content with undefined or empty content, like an empty string?

Reply:
Yeah, but the current ViewBox component doesn't handle children in a "real" way, it has some stuff passed in and it was using that to determine if it should show the incognito icon. The new parameter I added is an explicit override, the fallback check that could be error-prone is the existing behavior.

I think it would be better if the viewbox took child elements and had explicitly controlled visibility display, but I didn't want to rewrite the whole component and all its uses along the way to improving the memo rendering.